cmake/leveldb.cmake

```diff
@@ -0,0 +1,94 @@
+# Copyright (c) 2023 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+# This file is part of the transition from Autotools to CMake. Once CMake
+# support has been merged we should switch to using the upstream CMake
+# buildsystem.
+
+include(CheckCXXSymbolExists)
+check_cxx_symbol_exists(F_FULLFSYNC "fcntl.h" HAVE_FULLFSYNC)
```

Comment:
We use this in `util/system.cpp` as well, makes sense to move it to introspection.

Reply:
But it uses `defined(F_FULLFSYNC)` (see https://github.com/bitcoin/bitcoin/pull/3000), not `defined(HAVE_FULLFSYNC)`.

Reply:
> But it uses `defined(F_FULLFSYNC)` (see [bitcoin#3000](https://github.com/bitcoin/bitcoin/pull/3000)), not `defined(HAVE_FULLFSYNC)`.

Looks like a bug, doesn't it?

Reply:
FWIW, on my macOS systems, the `F_FULLFSYNC` is defined after `#include <fcntl.h>`.

Reply:
Yeah, this is kinda weird in our code so we can worry about it later.